### PR TITLE
fix: validator test checks based in diffs

### PR DESCRIPTION
**Pull request by sotnikov-s**
this PR:
- adds a new query `last_ack_seq_id` to the validator contract that returns the channel's sequence ID of the last received ack;
- refactors query deposit deduction and return checks: replaces concrete expected values with diff checks;
- adds the `last_ack_seq_id` usage to the await pieces of script which handles acknowledgements occurrence;
- cleans up some `echo`s, improves some steps.

As a result, it takes care of the following issues:

telegram-cloud-photo-size-2-5458446603770773456-y

telegram-cloud-photo-size-2-5458700066970779736-y

### CODE REVIEW

File: validator_test.sh [hunk 1]
```diff
@@ -139,15 +138,15 @@ TX_QUERY_REG_TX_HASH=$(echo $RES | jq --raw-output '.txhash')
 echo "TX Query ID: $TX_QUERY_ID, TX hash: $TX_QUERY_REG_TX_HASH"
 
 echo ""
-echo -n "Checking that deposits have been deducted completely from contract balance… "
+echo -n "Checking that the deposit have been deducted completely from contract balance… "
+
 RES=$(${BIN} query bank balances ${CONTRACT_ADDRESS} --output json --node ${NODE_URL})
-BALANCE=$(echo $RES | jq --raw-output '[.balances[] | select(.denom == "untrn")][0].amount')
-if [[ $BALANCE -eq 20000 ]]; then
+BALANCE_AFTER_ICQ_REG=$(echo $RES | jq --raw-output '[.balances[] | select(.denom == "untrn")][0].amount')
```
Comment (albertandrejev) [inline validator_test.sh:144]: I think we can add small delay before querying for balance update (5-7 seconds).

**Review by albertandrejev — APPROVED** [review]

File: validator_test.sh (hunk 1 above)
Comment (sotnikov-s) [reply validator_test.sh:144]: is it needed given the --broadcast-mode=block used in the above txs?

Comment (albertandrejev) [reply validator_test.sh:144]: sometimes it fails but still sits in the the mempool and waiting to be included into block.

Comment (albertandrejev) [reply validator_test.sh:144]: It will add more stability to the script imho.

**Review by albertandrejev — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Review by albertandrejev — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Review by albertandrejev — APPROVED** [review]
Tested ACK

**Review by quasisamurai — APPROVED** [review]
nice!

**Review by swelf19 — APPROVED** [review]

**Pull request merged**